ansible/roles/test/tasks/snmp/memory.yml

```diff
@@ -0,0 +1,105 @@
+      connection: local
+
+    - name: Check total memory via shell
+      shell: grep MemTotal /proc/meminfo | awk '{print $2}'
+      register: shell_total_memory
+      become: yes
+
+    - name: 'Validating SNMP total memory matches shell "/proc/meminfo" result'
+      assert:
+        that: "{{ (ansible_sysTotalMemery|int - shell_total_memory.stdout|int)|abs }} == 0"
+
+    - name: Gathering basic snmp facts about the device
+      snmp_facts: host={{ ansible_host }} version=v2c community={{ snmp_rocommunity }}
+      connection: local
+
+    - name: Check total free memory via shell
+      shell: grep MemFree /proc/meminfo | awk '{print $2}'
+      register: shell_total_free_memory
+      become: yes
+
+    - name: Validating SNMP total free memory matches shell result before stress test
+      assert:
+        that: "{{ '%.6f'|format((ansible_sysTotalFreeMemery|int - shell_total_free_memory.stdout|int)|abs / ansible_sysTotalFreeMemery|int) }} <= 0.03"
+
+    - set_fact: free_memery_before_test="{{ ansible_sysTotalFreeMemery }}"
+
+    - set_fact: test_momory="{{ ((ansible_sysTotalFreeMemery - 512000) / 1024)|int }}"
+
+    - name: Start memory stress generation
+      shell: python /tmp/memory.py {{ test_momory }} &
+      become: yes
+
+    - name: Wait for memory stress to reflect in SNMP
+      pause: seconds=20
+
+    - name: Gathering basic snmp facts about the device
+      snmp_facts: host={{ ansible_host }} version=v2c community={{ snmp_rocommunity }}
+      connection: local
+
+    - name: Check total free memory via shell
+      shell: grep MemFree /proc/meminfo | awk '{print $2}'
+      register: shell_total_free_memory
+      become: yes
+
+    - name: Validating SNMP total free memory matches shell result in stress test
+      assert:
+        that: "{{ '%.6f'|format((ansible_sysTotalFreeMemery|int - shell_total_free_memory.stdout|int)|abs / ansible_sysTotalFreeMemery|int) }} <= 0.03"
+
+    - name: Stop memory stress generation
+      shell: killall python /tmp/memory.py
+      become: yes
+      ignore_errors: true
+
+    - name: Wait for memory recovery to reflect in SNMP
+      pause: seconds=20
+
+    - name: Gathering basic snmp facts about the device
+      snmp_facts: host={{ ansible_host }} version=v2c community={{ snmp_rocommunity }}
+      connection: local
+
+    - name: Check total free memory via shell
+      shell: grep MemFree /proc/meminfo | awk '{print $2}'
+      register: shell_total_free_memory
+      become: yes
+
+    - name: Validating SNMP total free memory matches shell result after stress test
+      assert:
+        that: "{{ '%.6f'|format((ansible_sysTotalFreeMemery|int - shell_total_free_memory.stdout|int)|abs / ansible_sysTotalFreeMemery|int) }} <= 0.03"
+
+    - name: Validating memory leak issue on DUT
+      assert:
+        that: "{{ '%.6f'|format((free_memery_before_test|int - ansible_sysTotalFreeMemery|int)|abs / free_memery_before_test|int) }} <= 0.03"
```

Comment:
where does this memory leak comes from? the python program?

Reply:
Yes, the memory which hold by this python process should be released after terminated it.

Reply:
it is hard to say, the other process might increase the usage of memory.

Reply:
checking memory leak is a hard problem, I am mainly worry about the false positive.

Reply:
hmmm, just remove this test case from this PR and I'll find better way to test the memory leak. Thanks for your info.